src/Features/CSharp/Portable/SplitOrMergeIfStatements/CSharpMergeConsecutiveIfStatementsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.SplitOrMergeIfStatements;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.SplitOrMergeIfStatements
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.MergeConsecutiveIfStatements), Shared]
+    internal sealed class CSharpMergeConsecutiveIfStatementsCodeRefactoringProvider
+        : AbstractMergeConsecutiveIfStatementsCodeRefactoringProvider<ExpressionSyntax>
+    {
+        protected override string IfKeywordText => SyntaxFacts.GetText(SyntaxKind.IfKeyword);
+
+        protected override bool IsApplicableSpan(SyntaxNode node, TextSpan span, out SyntaxNode ifStatementNode)
```

Comment:
There used to be `TIfStatementSyntax` but I removed it when adding support for splitting VB ElseIf. Do you think it would be OK to have a type parameter called `TIfStatementSyntax` (that sounds like it would confusing to me).

Reply:
> it would be OK to have a type parameter called TIfStatementSyntax that would actually be SyntaxNode in VB? (That sounds a little confusing to me.)

Yup.  We have precedence for it elsewhere.  

You could also call it TIfOrElseIfStatementSyntax (if htat' actually what it represents).

Reply:
It is, but I'm treating an "else-if" as a kind of an "if statement" everywhere as a naming convention.

Reply:
TIfLikeStatementSyntax

Reply:
Most of the code that would benefit from that was moved into `IIfLikeStatementGenerator`. Adding a type parameter would no longer be useful.